src/index.js

```diff
@@ -42,39 +47,49 @@ export default class ReactPdfJs extends Component {
       cMapPacked,
     } = this.props;
     PdfJsLib.GlobalWorkerOptions.workerSrc = '//cdnjs.cloudflare.com/ajax/libs/pdf.js/2.0.943/pdf.worker.js';
-    PdfJsLib.getDocument({ url: file, cMapUrl, cMapPacked }).then((pdf) => {
+    this.PDFDocumentLoadingTask = PdfJsLib.getDocument({ url: file, cMapUrl, cMapPacked });
+    this.PDFDocumentLoadingTask.then((pdf) => {
       this.setState({ pdf });
       if (onDocumentComplete) {
         onDocumentComplete(pdf._pdfInfo.numPages); // eslint-disable-line
       }
-      pdf.getPage(page).then(p => this.drawPDF(p));
-    });
+      pdf.getPage(page).then(p => this.drawPDF(p), this.onDrawError);
+    },
+    this.onLoadError);
   }
 
   componentWillReceiveProps(newProps) {
     const { page, scale } = this.props;
     const { pdf } = this.state;
 
-    if (newProps.page !== page) {
-      pdf.getPage(newProps.page).then(p => this.drawPDF(p));
-    }
-    if (newProps.scale !== scale) {
-      pdf.getPage(newProps.page).then(p => this.drawPDF(p));
+    if (newProps.page !== page || newProps.scale !== scale) {
+      pdf.getPage(newProps.page).then(p => this.drawPDF(p), this.onDrawError);
     }
   }
 
-  getMaxScale = (page) => {
+  componentWillUnmount() {
+    if (this.PDFDocumentLoadingTask) this.PDFDocumentLoadingTask.destroy();
+  }
+
+  onLoadError = () => {
+    console.warn('Did not complete PDF Load');
+  }
+
+  onDrawError = () => {
+    console.warn('Did not draw PDF');
```

Comment:
these should be throws 😄

Reply:
ok, i'd then recommend we throw an error boundary around it at the Project level, that good?